vowpalwabbit/core/src/reductions/baseline_challenger_cb.cc

```diff
@@ -215,8 +217,7 @@ VW::LEARNER::base_learner* VW::reductions::baseline_challenger_cb_setup(VW::setu
 
   if (!options.was_supplied("cb_adf")) { THROW("cb_challenger requires cb_explore_adf or cb_adf"); }
 
-  bool emit_metrics = options.was_supplied("extra_metrics");
-  auto data = VW::make_unique<baseline_challenger_data>(emit_metrics, alpha, tau);
+  auto data = VW::make_unique<baseline_challenger_data>(all.global_metrics.are_metrics_enabled(), alpha, tau);
```

Comment:
Rather than reductions accessing via the all pointer, we could expose the metrics_manager via the `setup_base_i` interface

Reply:
Is this necessary? Seems like that could bloat this interface

Reply:
Generally we don't want to force reductions to use fields from all in their setup (or impl), but this can be fixed later if it seems important enough. This works for now